NuRadioMC/EvtGen/generator.py

```diff
@@ -1279,60 +1296,56 @@ def generate_eventlist_cylinder(filename, n_events, Emin, Emax,
             # we need to be careful to not double cound events. electron CC interactions apear twice in the event list
             # because of the two distinct showers that get created. Because second interactions are only calculated
             # for mu and tau cc interactions, this is not a problem.
-            mask_tau_cc = (data_sets["interaction_type"] == 'cc') & (np.abs(data_sets["flavors"]) == 16)
-            mask_mu_cc = (data_sets["interaction_type"] == 'cc') & (np.abs(data_sets["flavors"]) == 14)
-            mask_leptons = mask_tau_cc | mask_mu_cc
+            mask_tracks = data_sets["interaction_type"] == 'cc' & np.abs(data_sets["flavors"]) != 12
 
             E_all_leptons = (1 - data_sets["inelasticity"]) * data_sets["energies"]
             lepton_codes = copy.copy(data_sets["flavors"])
-            lepton_codes[lepton_codes == 14] = 13
-            lepton_codes[lepton_codes == -14] = -13
-            lepton_codes[lepton_codes == 16] = 15
-            lepton_codes[lepton_codes == -16] = -15
+            
+            # convert neutrino flavors (makes only sense for cc interaction which is ensured with "mask_leptons")
+            lepton_codes = lepton_codes - 1 * np.sign(lepton_codes)
```

Comment:
same here, better to it explicit as it was. The new code is harder to understand. 
I know that you added a check at the beginning but a likely future scenario is that someone might add another intial particle type and forgets about the implicit assumptions that are done further below in the cod.e

Reply:
I do not really see your point here. It should be quite easy to understand, is it really that confusing? And I have a hard time to see that one would add another initial particle type which is no neutrino to this function (without heavy rewriting it), it regardless of the above line specific to neutrinos. And even if one would accommodate a different particle type, is that desired if it makes the code harder to understand (because more  generic). We have already a different function to generate muon events

Reply:
@cg-laser do you agree with that or do you want that to be changed?